backend/services/AuthService.js

```diff
@@ -0,0 +1,59 @@
+const axios = require('axios');
+const jwt = require('jsonwebtoken');
+const logger = require('../config/logger');
+const UserService = require('./UserService');
+
+const client_id = process.env.GITHUB_CLIENT_ID;
+const client_secret = process.env.GITHUB_CLIENT_SECRET;
+const jwtSecret = process.env.JWT_SECRET;
+const scope = "read:user";
+
+if(!client_id){
```

Comment:
May we please move the if checks to be after the class declaration as it's standard practice for classes to be declared at the top

Reply:
the if checks are not part of the class that's why